xbmc/filesystem/ShoutcastFile.cpp

```diff
@@ -64,14 +64,19 @@ bool CShoutcastFile::Open(const CURL& url)
     url2.SetProtocol("http");
 
   bool result = m_file.Open(url2);
+  std::string icyTitle;
   if (result)
   {
-    m_tag.SetTitle(m_file.GetHttpHeader().GetValue("icy-name"));
-    if (m_tag.GetTitle().empty())
-      m_tag.SetTitle(m_file.GetHttpHeader().GetValue("ice-name")); // icecast
+    icyTitle = m_file.GetHttpHeader().GetValue("icy-name");
+    if (icyTitle.empty())
+      icyTitle = m_file.GetHttpHeader().GetValue("ice-name"); // icecast
     m_tag.SetGenre(m_file.GetHttpHeader().GetValue("icy-genre"));
     if (m_tag.GetGenre().empty())
       m_tag.SetGenre(m_file.GetHttpHeader().GetValue("ice-genre")); // icecast
+    // Handle no title or badly set up servers
+    if (icyTitle.empty() || (icyTitle == "This is my server name"))
```

Comment:
where does this magic string come from?

Reply:
It comes from stations that were set up by lazy people who haven't bothered to change the title header from the default setting.  I've encountered it several times during testing.

Reply:
A comment would help here explaining that bad servers often have this junk data